### PR TITLE
Add new prefix logging and jsonification option

### PR DESCRIPTION
**Pull request by Spring3**
Caller module prefix is now being also logged out
Support for metadata -> json  transformation

### CODE REVIEW

**Review by jkernech — CHANGES_REQUESTED** [review]
console.log needs to be removed, aside from that LGTM

File: src/index.js [hunk 1]
```diff
@@ -82,10 +82,18 @@ class Logstash extends stream.Subscriber {
...
-          `[${prefix.timestamp}${prefix.environment}${prefix.logLevel}${prefix.reqId}]${content.text}` :
-          content.text;
-        this.winston.log(messageLevel, messageText, content.meta);
+        let prefixText = !prefix.isEmpty ?
+          `[${prefix.timestamp}${prefix.environment}${prefix.logLevel}${prefix.reqId}] ` : '';
+        // if prefix contains these props, then caller module prefix was configured by settings/env
+        if ({}.hasOwnProperty.call(prefix, 'module') &&
+            {}.hasOwnProperty.call(prefix, 'function') &&
+            {}.hasOwnProperty.call(prefix, 'project')) {
+          prefixText += `[${prefix.project}${prefix.module}${prefix.function}] `;
```
Comment (mrister) [inline src/index.js:91]: shouldn't prefix project and function be separated by `:` what would a smaple output look like ?

Comment (Spring3) [reply src/index.js:91]: It is done internally inside logtify's main module. All that is needed - is just to position them as you wish

**Review by jkernech — APPROVED** [review]

**Review by jkernech — CHANGES_REQUESTED** [review]
Tests need to be fixed

File: Makefile [hunk 2]
```diff
@@ -8,10 +8,10 @@ lint:
 
 test:
 	make lint
-	make cover
+	npm run test
 	
 cover:
-	node_modules/.bin/istanbul cover  node_modules/.bin/_mocha -- test --recursive --timeout=3000
+	make test
```
Comment (jkernech) [inline Makefile]: Sorry but something seems a bit off here, especially the mix with `make` and `npm` commands, here is a good example taken from one of our repo :

```
lint:
	npm run lint

test:
	make lint
	make cover

cover:
	node_modules/.bin/istanbul cover node_modules/.bin/_mocha -- test --recursive
```

**Review by mrister — APPROVED** [review]

**Review by jkernech — APPROVED** [review]

**Review by ykhatal — APPROVED** [review]

**Pull request merged**